Build failure: badkeys

### Describe the bug

The `badkeys` python package fails its tests on `nixos-unstable`. Specifically, the `tests/test_rsainvalid.py::test_rsainvalid` test fails with an AssertionError. This is likely due to recent updates to its `cryptography` dependency in `nixpkgs`.

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix build --impure --expr '(import <nixpkgs> {}).badkeys'`

### Expected behavior

Tests should pass successfully. 

### Additional context

```
> =================================== FAILURES ===================================
> ________________________ TestRSAInvalid.test_rsainvalid ________________________
>
> self = <tests.test_rsainvalid.TestRSAInvalid testMethod=test_rsainvalid>
>
>     def test_rsainvalid(self):
>         with open(f"{TDPATH}rsa-e1.key") as f:
>             key = f.read()
>         r = badkeys.checkpubkey(key, checks=["rsainvalid"])
> >       self.assertTrue("rsainvalid" in r["results"])
> E       AssertionError: False is not true
>
> tests/test_rsainvalid.py:14: AssertionError
```

### Notify maintainers

@getchoo @NixOS/ngi

--- a/pkgs/by-name/ba/badkeys/package.nix
+++ b/pkgs/by-name/ba/badkeys/package.nix
@@ -4,6 +4,7 @@
   fetchFromGitHub,
   python3Packages,
   testers,
+  fetchpatch2,
 }:
 
 python3Packages.buildPythonApplication (finalAttrs: {
@@ -17,6 +18,20 @@ python3Packages.buildPythonApplication (finalAttrs: {
     tag = "v${finalAttrs.version}";
     hash = "sha256-i6wKe3djZL2tHjq3UrC+edU6Gz8EYSXSUry61BaI0v4=";
   };
+
+  patches = [
+    # TODO: remove on version > 0.0.19
+    (fetchpatch2 {
+      name = "Adapt-tests-for-rsainvalid-check-for-latest-Python-c.patch";
+      url = "https://github.com/badkeys/badkeys/commit/0bdcc68378bf389aebcc714b8c24de1c8a88a4e9.patch?full_index=1";
+      hash = "sha256-K0AsRNxfDQ/qC7abt0rr8tKNP9cfvWJ75DscdyoxF08=";
+    })
+    (fetchpatch2 {
+      name = "Add-test-for-Diffie-Hellman-DH-key.patch";
+      url = "https://github.com/badkeys/badkeys/commit/4e16c47dc79dd98865f4e3ec4b981306630a57a3.patch?full_index=1";
+      hash = "sha256-6F4lfUOmSkCF4XuiPwnJGD2RrxT5/jsillAyW9vzXMw=";
+    })
+  ];
 
   build-system = with python3Packages; [
     setuptools

